File: problems/Best Time to Buy and Sell Stock/maxProfit.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        var minPrice = Int32.MaxValue;
        var maxProfit = 0;

        foreach (var item in prices) {
            minPrice = Math.Min(minPrice, item);
            maxProfit = Math.Max(maxProfit, item - minPrice);
        }

        return maxProfit;
    }
}

File: problems/Best Time to Buy and Sell Stock/maxProfit.java
class Solution {
    public int maxProfit(int[] prices) {
        int minPrice = Integer.MAX_VALUE;
        int maxProfit = 0;

        for (int item : prices) {
            minPrice = Math.min(minPrice, item);
            maxProfit = Math.max(maxProfit, item - minPrice);
        }

        return maxProfit;
    }
}
